### PR TITLE
VIH-10475 A new participant being added resets judges custom display name

### PR DESCRIPTION
**Pull request by will-craig**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/VIH-10475

### Change description ###
Updated displayname update to also send a participantUpdated event to other participants to show new display name

### CODE REVIEW

File: VideoWeb/VideoWeb/Controllers/ParticipantsController.cs [hunk 1]
```diff
@@ -137,14 +137,13 @@ public async Task<IActionResult> GetHeartbeatDataForParticipantAsync(Guid confer
...
-            [FromBody] UpdateParticipantDisplayNameRequest participantRequest)
+        public async Task<IActionResult> UpdateParticipantDisplayNameAsync(Guid conferenceId, Guid participantId, [FromBody] UpdateParticipantDisplayNameRequest participantRequest)
         {
             try
             {
-                var apiRequest = _mapperFactory.Get<UpdateParticipantDisplayNameRequest, UpdateParticipantRequest>()
-                    .Map(participantRequest);
-                await _videoApiClient.UpdateParticipantDetailsAsync(conferenceId, participantId, apiRequest);
+                var apiRequest = _mapperFactory.Get<UpdateParticipantDisplayNameRequest, UpdateParticipantRequest>().Map(participantRequest);
+                _videoApiClient.UpdateParticipantDetailsAsync(conferenceId, participantId, apiRequest).Wait(); //wait for update to complete before fetching participants to notify
```
Comment (oliver-scott) [inline VideoWeb/VideoWeb/Controllers/ParticipantsController.cs:145]: can we not use `await` here instead?

`await _videoApiClient.UpdateParticipantDetailsAsync(conferenceId, participantId, apiRequest);`

File: VideoWeb/VideoWeb/Controllers/ParticipantsController.cs [hunk 2]
```diff
@@ -165,6 +179,7 @@ public async Task<IActionResult> GetHeartbeatDataForParticipantAsync(Guid confer
         [HttpGet("{conferenceId}/vhofficer/participants")]
         [ProducesResponseType(typeof(IEnumerable<ParticipantContactDetailsResponseVho>), (int)HttpStatusCode.OK)]
         [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
```
Comment (oliver-scott) [inline VideoWeb/VideoWeb/Controllers/ParticipantsController.cs:182]: not sure a second BadRequest annotation is needed here

File: VideoWeb/VideoWeb.UnitTests/Controllers/ParticipantController/UpdateJudgeDisplayNameTests.cs [hunk 3]
```diff
@@ -13,9 +14,14 @@
...
 using EventHubEventType = VideoWeb.EventHub.Enums.EventType;
 using Autofac.Extras.Moq;
+using VideoApi.Contract.Enums;
 using VideoWeb.Contract.Request;
 using VideoWeb.Mappings;
 using VideoApi.Contract.Requests;
+using VideoApi.Contract.Responses;
+using VideoWeb.Contract.Responses;
+using VideoWeb.EventHub.Models;
+using LinkedParticipantResponse = VideoWeb.Contract.Responses.LinkedParticipantResponse;
```
Comment (oliver-scott) [inline VideoWeb/VideoWeb.UnitTests/Controllers/ParticipantController/UpdateJudgeDisplayNameTests.cs:24]: can we remove this, doesn't look like it is used

`using LinkedParticipantResponse = VideoWeb.Contract.Responses.LinkedParticipantResponse;`

**Review by oliver-scott — COMMENTED** [review]
looks good, just a few comments

File: VideoWeb/VideoWeb/Controllers/ParticipantsController.cs (hunk 1 above)
Comment (will-craig) [reply VideoWeb/VideoWeb/Controllers/ParticipantsController.cs:145]: I used wait, as the next statement fetches the updated conference data, and wanted to be sure this request completes before call ing the get request to retrieve the updated conference data

Comment (will-craig) [reply VideoWeb/VideoWeb/Controllers/ParticipantsController.cs]: Actually, I think you're right, shouldn't make a difference actually if it awaits the task completion before then next await. Was thinking about it in the wrong way

Comment (oliver-scott) [reply VideoWeb/VideoWeb/Controllers/ParticipantsController.cs]: but I think await should wait for the completion too. Just a bit conscious that `Wait()` could lead to deadlocks as it blocks the thread.

does await not work?

Comment (oliver-scott) [reply VideoWeb/VideoWeb/Controllers/ParticipantsController.cs]: sorry sent that last message at the last time as yours, disregard :)

**Review by oliver-scott — APPROVED** [review]

**Review by komaldabhi1509 — APPROVED** [review]

**Pull request merged**